app/src/main/java/org/mozilla/tv/firefox/webrender/cursor/CursorViewModel.kt

```diff
@@ -60,9 +67,61 @@ class CursorViewModel(
      * manipulation work to [CursorViewModel]
      */
     fun onSelectKeyEvent(action: Int, pos: PointF) {
-        val now = SystemClock.uptimeMillis()
-        MotionEvent.obtain(now - DOWN_TIME_OFFSET_MILLIS, now, action, pos.x, pos.y, 0).use {
+        validateMotionEvent(action, SystemClock.uptimeMillis(), pos)?.use {
             _touchSimulationLiveData.value = Consumable.from(it)
         }
     }
+
+    /**
+     * Problem: there is a distinctive time difference between [MotionEvent.ACTION_DOWN] and
+     * [MotionEvent.ACTION_UP] when using DPAD_SELECT_KEY. [GeckoSession] requires them to have matching
+     * [MotionEvent.getEventTime] in order to register such [KeyEvent] as "Click" touch event.
+     *
+     * [validateMotionEvent] checks for previous ACTION_DOWN vs. upcoming ACTION_UP to calculate
+     * the diff against [ViewConfiguration.getLongPressTimeout] to determine whether "Click" or
+     * "Long Press" event. Also filters unnecessary [MotionEvent] that occurs when holding down
+     * DPAD_SELECT_KEY (see (*)) by bypassing consecutive ACTION_DOWN calls.
+     *
+     * (*) Capture the first instance of [KeyEvent.ACTION_DOWN] upon long press and check against
+     * the first instance of [KeyEvent.ACTION_UP].
+     */
+    @VisibleForTesting
+    private fun validateMotionEvent(action: Int, now: Long, pos: PointF) : MotionEvent? {
+        var curr = MotionEvent.obtain(now - DOWN_TIME_OFFSET_MILLIS, now, action, pos.x, pos.y, 0)
+
+        // ACTION_DOWN == 0, ACTION_UP == 1
+        Log.d("ValidatedMotionEvent", String.format("curr: %d, prev: %d", action, prevDownMotionEvent?.action))
+
+        when (action) {
+            KeyEvent.ACTION_DOWN -> {
+                if (prevDownMotionEvent == null) {
+                    prevDownMotionEvent = MotionEvent.obtain(curr)
+                } else {
+                    return null
+                }
```

Comment:
Yes that's right, so then we ignore by returning null

Reply:
In hindsight that could potentially be a weird edge case where `ACTION_UP` never arrives then we should probably reset

Reply:
I'm going to write a test case for this

Reply:
Yeah, though instead of resetting after a timeout or something, always overwriting the previous down event would be simpler - we'll just a get a newer timestamp, and won't have old events hanging around.